### ISSUE TITLE
qt5 fails to build on i686-w64-mingw32.static.posix

### ISSUE THREAD

**Issue by uwehermann** [mxe/mxe#1347]
After commenting out / disabling mysql support in qt5 (see bug #1346) the qt5 build now fails with the following error:

```
/home/uwe/mxe-git/usr/i686-w64-mingw32.static.posix/lib/libpthread.a(libwinpthread_la-mutex.o): In function `pthread_mutex_lock':
/home/uwe/mxe-git/tmp-gcc-i686-w64-mingw32.static.posix/gcc-4.9.3/mingw-w64-v4.0.6/mingw-w64-libraries/winpthreads/src/mutex.c:182: multiple definition of `pthread_mutex_lock'
/home/uwe/mxe-git/usr/i686-w64-mingw32.static.posix/lib/libpq.a(pthread-win32.o):pthread-win32.c:(.text+0x70): first defined here
/home/uwe/mxe-git/usr/i686-w64-mingw32.static.posix/lib/libpthread.a(libwinpthread_la-mutex.o): In function `pthread_mutex_unlock':
/home/uwe/mxe-git/tmp-gcc-i686-w64-mingw32.static.posix/gcc-4.9.3/mingw-w64-v4.0.6/mingw-w64-libraries/winpthreads/src/mutex.c:202: multiple definition of `pthread_mutex_unlock'
/home/uwe/mxe-git/usr/i686-w64-mingw32.static.posix/lib/libpq.a(pthread-win32.o):pthread-win32.c:(.text+0xa0): first defined here
/home/uwe/mxe-git/usr/i686-w64-mingw32.static.posix/lib/libpthread.a(libwinpthread_la-mutex.o): In function `pthread_mutex_init':
/home/uwe/mxe-git/tmp-gcc-i686-w64-mingw32.static.posix/gcc-4.9.3/mingw-w64-v4.0.6/mingw-w64-libraries/winpthreads/src/mutex.c:248: multiple definition of `pthread_mutex_init'
/home/uwe/mxe-git/usr/i686-w64-mingw32.static.posix/lib/libpq.a(pthread-win32.o):pthread-win32.c:(.text+0x30): first defined here
/home/uwe/mxe-git/usr/i686-w64-mingw32.static.posix/lib/libpthread.a(libwinpthread_la-thread.o): In function `pthread_getspecific':
/home/uwe/mxe-git/tmp-gcc-i686-w64-mingw32.static.posix/gcc-4.9.3/mingw-w64-v4.0.6/mingw-w64-libraries/winpthreads/src/thread.c:752: multiple definition of `pthread_getspecific'
/home/uwe/mxe-git/usr/i686-w64-mingw32.static.posix/lib/libpq.a(pthread-win32.o):pthread-win32.c:(.text+0x20): first defined here
/home/uwe/mxe-git/usr/i686-w64-mingw32.static.posix/lib/libpthread.a(libwinpthread_la-thread.o): In function `pthread_setspecific':
/home/uwe/mxe-git/tmp-gcc-i686-w64-mingw32.static.posix/gcc-4.9.3/mingw-w64-v4.0.6/mingw-w64-libraries/winpthreads/src/thread.c:765: multiple definition of `pthread_setspecific'
/home/uwe/mxe-git/usr/i686-w64-mingw32.static.posix/lib/libpq.a(pthread-win32.o):pthread-win32.c:(.text+0x10): first defined here
/home/uwe/mxe-git/usr/i686-w64-mingw32.static.posix/lib/libpthread.a(libwinpthread_la-thread.o): In function `pthread_self':
/home/uwe/mxe-git/tmp-gcc-i686-w64-mingw32.static.posix/gcc-4.9.3/mingw-w64-v4.0.6/mingw-w64-libraries/winpthreads/src/thread.c:912: multiple definition of `pthread_self'
/home/uwe/mxe-git/usr/i686-w64-mingw32.static.posix/lib/libpq.a(pthread-win32.o):pthread-win32.c:(.text+0x0): first defined here
collect2: error: ld returned 1 exit status
Makefile.Release:67: recipe for target 'release/test-qt5.exe' failed
make[3]: *** [release/test-qt5.exe] Error 1
make[3]: Leaving directory '/home/uwe/mxe-git/tmp-qtbase-i686-w64-mingw32.static.posix/qtbase-opensource-src-5.6.0/test-qt'
Makefile:34: recipe for target 'release' failed
make[2]: *** [release] Error 2
make[2]: Leaving directory '/home/uwe/mxe-git/tmp-qtbase-i686-w64-mingw32.static.posix/qtbase-opensource-src-5.6.0/test-qt'
Makefile:548: recipe for target 'build-only-qtbase_i686-w64-mingw32.static.posix' failed
make[1]: *** [build-only-qtbase_i686-w64-mingw32.static.posix] Error 2
make[1]: Leaving directory '/home/uwe/mxe-git'

```

**Comment by uwehermann** [comment]
Commenting out the whole qt test chunk seem to lead to a working qt5/qtbase* build on i686-w64-mingw32.static.posix.

```
-    mkdir            '$(1)/test-qt'
-    cd               '$(1)/test-qt' && '$(PREFIX)/$(TARGET)/qt5/bin/qmake' '$(PWD)/src/qt-test.pro'
-    $(MAKE)       -C '$(1)/test-qt' -j '$(JOBS)'
-    $(INSTALL) -m755 '$(1)/test-qt/release/test-qt5.exe' '$(PREFIX)/$(TARGET)/bin/'
+    # mkdir            '$(1)/test-qt'
+    # cd               '$(1)/test-qt' && '$(PREFIX)/$(TARGET)/qt5/bin/qmake' '$(PWD)/src/qt-test.pro'
+    # $(MAKE)       -C '$(1)/test-qt' -j '$(JOBS)'
+    # $(INSTALL) -m755 '$(1)/test-qt/release/test-qt5.exe' '$(PREFIX)/$(TARGET)/bin/'

-    # build test the manual way
-    mkdir '$(1)/test-$(PKG)-pkgconfig'
-    '$(PREFIX)/$(TARGET)/qt5/bin/uic' -o '$(1)/test-$(PKG)-pkgconfig/ui_qt-test.h' '$(TOP_DIR)/src/qt-test.ui'
-    '$(PREFIX)/$(TARGET)/qt5/bin/moc' \
-        -o '$(1)/test-$(PKG)-pkgconfig/moc_qt-test.cpp' \
-        -I'$(1)/test-$(PKG)-pkgconfig' \
-        '$(TOP_DIR)/src/qt-test.hpp'
-    '$(PREFIX)/$(TARGET)/qt5/bin/rcc' -name qt-test -o '$(1)/test-$(PKG)-pkgconfig/qrc_qt-test.cpp' '$(TOP_DIR)/src/qt-test.qrc'
-    '$(TARGET)-g++' \
-        -W -Wall -Werror -std=c++0x -pedantic \
-        '$(TOP_DIR)/src/qt-test.cpp' \
-        '$(1)/test-$(PKG)-pkgconfig/moc_qt-test.cpp' \
-        '$(1)/test-$(PKG)-pkgconfig/qrc_qt-test.cpp' \
-        -o '$(PREFIX)/$(TARGET)/bin/test-$(PKG)-pkgconfig.exe' \
-        -I'$(1)/test-$(PKG)-pkgconfig' \
-        `'$(TARGET)-pkg-config' Qt5Widgets --cflags --libs`
+    # # build test the manual way
+    # mkdir '$(1)/test-$(PKG)-pkgconfig'
+    # '$(PREFIX)/$(TARGET)/qt5/bin/uic' -o '$(1)/test-$(PKG)-pkgconfig/ui_qt-test.h' '$(TOP_DIR)/src/qt-test.ui'
+    # '$(PREFIX)/$(TARGET)/qt5/bin/moc' \
+    #     -o '$(1)/test-$(PKG)-pkgconfig/moc_qt-test.cpp' \
+    #     -I'$(1)/test-$(PKG)-pkgconfig' \
+    #     '$(TOP_DIR)/src/qt-test.hpp'
+    # '$(PREFIX)/$(TARGET)/qt5/bin/rcc' -name qt-test -o '$(1)/test-$(PKG)-pkgconfig/qrc_qt-test.cpp' '$(TOP_DIR)/src/qt-test.qrc'
+    # '$(TARGET)-g++' \
+    #     -W -Wall -Werror -std=c++0x -pedantic \
+    #     '$(TOP_DIR)/src/qt-test.cpp' \
+    #     '$(1)/test-$(PKG)-pkgconfig/moc_qt-test.cpp' \
+    #     '$(1)/test-$(PKG)-pkgconfig/qrc_qt-test.cpp' \
+    #     -o '$(PREFIX)/$(TARGET)/bin/test-$(PKG)-pkgconfig.exe' \
+    #     -I'$(1)/test-$(PKG)-pkgconfig' \
+    #     `'$(TARGET)-pkg-config' Qt5Widgets --cflags --libs`
```

**Comment by tonytheodore** [comment]
Try applying the patches from #1009 for mysql and postgresql.

**Comment by tonytheodore** [comment]
See https://github.com/mxe/mxe/wiki/Package-Specific-Notes#toolchains-with-c11-features-ie--w64-mingw32staticposix. Can you try:
```
make ... qtbase_CONFIGURE_OPTS='-no-sql-psql -no-sql-mysql' libmysqlclient_BUILD= postgresql_BUILD=
```

**Comment by uwehermann** [comment]
This worked for me upon a fresh clone (plus patches for unrelated gcc6 issues):

```
make -j 4 JOBS=4 gcc qt5 qtbase_CONFIGURE_OPTS='-no-sql-psql -no-sql-mysql' libmysqlclient_BUILD= postgresql_BUILD=
```

**Comment by tonytheodore** [comment]
@uwehermann see #1488 for a proper fix.

**Comment by tonytheodore** [comment]
fixed in #1488
